crutest/src/main.rs

```diff
@@ -1968,6 +2046,353 @@ async fn replay_workload(
+            )
+            .await
+        {
+            Ok(ReplaceResult::Started) => {}
+            x => {
+                bail!("Failed replace: {:?}", x);
+            }
+        }
+
+        info!(log, "[{c}] Wait for activation after replacement");
+        loop {
+            let is_active = guest.query_is_active().await.unwrap();
+            if is_active {
+                break;
+            }
+            info!(
+                log,
+                "[{c}] Upstairs query_is_active reports: {:?}", is_active
+            );
+            tokio::time::sleep(tokio::time::Duration::from_secs(10)).await;
+        }
+
+        // Our activate request task should have ended.
+        if let Err(e) = handle.await.unwrap() {
+            bail!("Failure reported in activate task {:?}", e);
+        }
+
+        info!(log, "[{c}] Verify volume after replacement");
+        if let Err(e) = verify_volume(guest, ri, false).await {
+            bail!("Requested volume verify failed: {:?}", e)
+        }
+
+        // Wait for all IO to finish before we continue
+        loop {
+            let wc = guest.show_work().await?;
+            info!(
+                log,
+                "[{c}] Jobs Up:{} Ds:{}  DS active:{}",
+                wc.up_count,
+                wc.ds_count,
+                wc.active_count
+            );
+            if wc.up_count + wc.ds_count == 0 && wc.active_count == 3 {
+                info!(log, "[{c}] Replay: All jobs finished, all DS active.");
+                break;
+            }
+            tokio::time::sleep(tokio::time::Duration::from_secs(4)).await;
+        }
+
+        old_ds = (old_ds + 1) % 4;
+        new_ds = (new_ds + 1) % 4;
+
+        c += 1;
+        match wtq {
+            WhenToQuit::Count { count } => {
+                if c > *count {
+                    break;
+                }
+            }
+            WhenToQuit::Signal { shutdown_rx } => {
+                match shutdown_rx.try_recv() {
+                    Ok(SignalAction::Shutdown) => {
+                        println!("shutting down in response to SIGUSR1");
+                        break;
+                    }
+                    Ok(SignalAction::Verify) => {
+                        println!("Verify Volume");
+                        if let Err(e) = verify_volume(guest, ri, false).await {
+                            bail!("Requested volume verify failed: {:?}", e)
+                        }
+                    }
+                    _ => {} // Ignore everything else
+                }
+            }
+        }
+    }
+
+    info!(log, "Test replace_while_reconcile has completed");
+    Ok(())
+}
+
+// Test that a downstairs can be replaced before activation when the original
+// downstairs is offline.
+// Make use of dsc to stop our downstairs before activation.
+// Do a fill on each loop so every extent will need to be repaired.
+async fn replace_before_active(
+    guest: &Arc<Guest>,
+    wtq: &mut WhenToQuit,
```

Comment:
Same question here

Reply:
Yes, both can be passed by value and I did that in 85fc1e20